### PR TITLE
Facility upsert: Deny others from updating a user's facility

### PR DESCRIPTION
**Pull request by afzalIbnSH**
Unless the other user is a superuser. Retains the existing owner in such cases.

### CODE REVIEW

**Comment by afzalIbnSH** [comment]
issue #91

**Comment by afzalIbnSH** [comment]
Screenshot 2020-03-25 at 1 30 25 PM

File: care/facility/api/serializers/facility.py [hunk 1]
```diff
@@ -64,22 +68,30 @@ def validate_phone_number(self, value):
...
         ).first()
 
+        user = self.context["user"]
         if not facility:
+            validated_data["created_by"] = user
             facility = Facility.objects.create(**validated_data)
         else:
+            if facility.created_by != user and not user.is_superuser:
+                raise PermissionDenied(
+                    f"{facility.name}, {DISTRICT_CHOICES[facility.district][1]} is owned by another user"
```
Comment (anroopak) [inline care/facility/api/serializers/facility.py]: Can we reuse `__str__` here?

**Comment by afzalIbnSH** [comment]
@vigneshhari @biswaz Please merge #90 before this. I'll have to modify the unit tests. I've deliberately left the unit test to break once this is merged so that it's not missed.

File: care/facility/api/serializers/facility.py (hunk 1 above)
Comment (afzalIbnSH) [reply care/facility/api/serializers/facility.py]: https://github.com/coronasafe/care/pull/93/commits/8c2f73be03285ff242a9ccff7bd500cac4826655

**Comment by vigneshhari** [comment]
@afzalIbnSH solve the conflicts and fix the unit tests so that it does not break

**Pull request merged**